### ISSUE TITLE
"Table of Content" outline: add skipped heading levels warnings

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1903]
The "Table of Content" outline introduced in #1896 (experimental, so far) is a pretty neat feature and could also be further improved. For now, it warns users when a heading is empty:

![28209234-efc09964-6891-11e7-999d-3c3ddb610b03](https://user-images.githubusercontent.com/1682452/28218319-75fd0fbe-68b8-11e7-9a1c-512ae3d60520.png)

For better accessibility, educating users, and **also for SEO**, it would be great if it also warnd users when heading levels are skipped. For inspiration, here's what the "developers toolbar" Firefox add-on does when analysing the document outline:

![screen shot 2017-07-14 at 13 01 02](https://user-images.githubusercontent.com/1682452/28218438-cf72d2fe-68b8-11e7-9d59-edebf0245a49.png)

See related [conversation on Slack](https://wordpress.slack.com/archives/C02QB2JS7/p1500030355721575) and also #1760

**Comment by mtias** [comment]
Adding the "good first task" as it should be quite a straightforward improvement.

**Comment by StaggerLeee** [comment]
Make this list anchored ? So clicking on it scroll User to specific block, and center, focus block.

**Comment by mtias** [comment]
@StaggerLeee yes, that is mentioned on the introduction of the feature.
